virtwho/virt/virt.py

```diff
@@ -705,16 +709,22 @@ def _send_data(self, data_to_send):
                             sources_erred.append(source_key)
                         retry = False  # Only retry on 429
 
+        # Was anything sent?
+        any_report_sent = all((source_key in sources_sent or source_key in sources_erred)
+                              for source_key in self.source_keys)
+
         # Terminate this thread if we have sent one report for each source
-        if all((source_key in sources_sent or source_key in sources_erred)
-               for source_key in self.source_keys) and self._oneshot:
-            if not self.options.print_:
-                self.logger.debug('At least one report for each connected '
-                                  'source has been sent. Terminating.')
-            else:
-                self.logger.debug('All info to print has been gathered. '
-                                  'Terminating.')
-            self.stop()
+        if any_report_sent is False:
```

Comment:
As mentioned above, this condition appears to only ever be false if `data_to_send` does not include a report for all of `self.source_keys`. In this case the `batch_host_guest_report` will not have the data from all sources but will be printed once more (as it is already printed during the call to hypervisorCheckIn).